checkov/github/runner.py

```diff
@@ -53,3 +54,22 @@ def require_external_checks(self) -> bool:
     def import_registry(self) -> BaseCheckRegistry:
         from checkov.github.registry import registry
         return registry
+
+    def _load_files(
+            self,
+            files_to_load: list[str],
+            filename_fn: Callable[[str], str] | None = None,
+    ) -> None:
+        super(Runner, self)._load_files(files_to_load=files_to_load, filename_fn=filename_fn)
+
+        for file_path, definition in self.definitions.items():
+            file_name = Path(file_path).stem
+            ckv_metadata = {
+                'file_name': file_name,
+                'org_complementary_metadata': self.github.org_complementary_metadata,
+                'repo_complementary_metadata': self.github.repo_complementary_metadata,
+            }
+            if isinstance(definition, dict):
+                definition[CKV_METADATA] = ckv_metadata
+            elif isinstance(definition, list):
+                definition.append(ckv_metadata)
```

Comment:
@marynaKK Why not use `force_list` and have 1 way to handle this, at all times?

Reply:
I didn't want to impact existing checks more than I already did in this pr.
Is a good idea actually  :)

Reply:
I tried to- this is mixing between data in form of a list and an addition to an existing list. 
I still will have if-else clauses, but that would be on modified data -> possible additional errors.

Reply:
So make it a dict always. Does that solve it?

Reply:
no- it will change the schema of the data, then I can no longer use the validate schema, or I would have to change back the data to the original schema.
why is this wrong to support both cases?

Reply:
@marynaKK Because it means multiple places across the code need to always take into account the structure of the data may be different. Data should be structured consistently. Please find a solution to this